operate/schema/src/main/java/io/camunda/operate/schema/IndexMappingDifference.java

```diff
@@ -150,11 +120,73 @@ public String toString() {
         + '}';
   }
 
-  public static IndexMappingProperty createIndexMapping(
-      final Entry<String, Object> propertiesMapEntry) {
-    return new IndexMappingProperty()
-        .setName(propertiesMapEntry.getKey())
-        .setTypeDefinition(propertiesMapEntry.getValue());
+  public static class IndexMappingDifferenceBuilder {
+    private boolean ignoreDynamicDifferences;
+    private IndexMapping left;
+    private IndexMapping right;
+
+    public static IndexMappingDifferenceBuilder builder() {
+      return new IndexMappingDifferenceBuilder();
+    }
+
+    public IndexMappingDifferenceBuilder setLeft(final IndexMapping left) {
+      this.left = left;
+      return this;
+    }
+
+    public IndexMappingDifferenceBuilder setRight(final IndexMapping right) {
+      this.right = right;
+      return this;
+    }
+
+    public IndexMappingDifferenceBuilder setIgnoreDynamicDifferences(
+        final boolean ignoreDynamicDifferences) {
+      this.ignoreDynamicDifferences = ignoreDynamicDifferences;
+      return this;
+    }
+
+    public IndexMappingDifference build() {
+      final MapDifference<String, Object> difference = Maps.difference(left.toMap(), right.toMap());
+      final IndexMappingDifference diff =
+          new IndexMappingDifference()
+              .setEqual(difference.areEqual())
+              .setEntriesOnlyOnLeft(
+                  difference.entriesOnlyOnLeft().entrySet().stream()
+                      .map(p -> createIndexMappingProperty(p))
+                      .collect(Collectors.toSet()))
+              .setEntriesOnlyOnRight(
+                  difference.entriesOnlyOnRight().entrySet().stream()
+                      .map(p -> createIndexMappingProperty(p))
+                      .collect(Collectors.toSet()))
+              .setEntriesInCommon(
+                  difference.entriesInCommon().entrySet().stream()
+                      .map(p -> createIndexMappingProperty(p))
+                      .collect(Collectors.toSet()));
+
+      if (ignoreDynamicDifferences
+          && (Boolean.parseBoolean(left.getDynamic())
+              || Boolean.parseBoolean(right.getDynamic()))) {
```

Comment:
Nitpicking: this could become an accessor in `IndexMapping`, e.g. `IndexMapping#IsDynamicTrue`. That way the conversion logic would be in one place. Not critical for getting this merged, for sure.

Reply:
Yeah I went back and forth on this whether to make it an accessor member of the IndexMapping class or not. In the end I thought it might be slightly confusing to have an accessor for the string dynamic field and a boolean isDynamic accessor and went with a utility function in the validator. However I completely agree with any arguments to be made for making it an accessor instead, and we could refactor this in a later cleanup.